### DISCUSSION TITLE
Get errors when using docker-compose, help needed

### DISCUSSION THREAD

**Discussion by LoganTann** [denodrivers/postgres#405] [Q&A]
Hello, I'm trying to run deno and postgres in a docker container using docker compose.

My deno application is using denodb, which uses this project to connect postgres.

I get this strange error :

`error: Uncaught (in promise) Error: Could not check if server accepts SSL connections, server responded with: 69`

after a reset, still the same error but response is different

`error: Uncaught (in promise) Error: Could not check if server accepts SSL connections, server responded with: 82`

I don't understand why it happens, any idea ?

Here's a snapshot of my code (interesting file might be the docker-compose.yml) https://github.com/LoganTann/redbeansoup-v2/tree/8c8a42e64ce7c0102b7da0729009a21b2d38db3b

operating system is windows 10.

config in deno (that seems totally correct regarding to the parameters set in the docker compose file) :
database: "redbeansoup",
host: "postgres",
username: "postgres",
password: "redbeansoup-bot",
port: 5432

**Comment by Soremwar** [comment]
Hi @LoganTann 

By looking at your docker configuration, I think it's likely that you are trying to use the DB before it is ready to accept queries. Try starting only the database service, check if it works with some database manager and then start one of the services that depend on it

If this turns out to be the problem, check the docker-compose file in this repo (pay special attention to the `wait` program) in order to wait for the database to initialize before everything else fires up

**Reply by LoganTann** [reply]
Hello, Even after waiting for the db, I get the same error

Here's the output of the postgres image : 

    db_1           | 2022-06-29 17:24:49.467 UTC [1] LOG:  database system is ready to accept connections
    db_1           | 2022-06-29 17:28:33.664 UTC [32] FATAL:  received unencrypted data after SSL request
    db_1           | 2022-06-29 17:28:33.664 UTC [32] DETAIL:  This could be either a client-software bug or evidence of an attempted man-in-the-middle attack.

and the output of deno

    dbTest_1       | error: Uncaught (in promise) Error: Could not check if server accepts SSL connections, server responded with: 69

The same error happens when I run the deno script without docker (but postgres remains inside a docker container)
